pandablocks/commands.py

```diff
@@ -229,49 +239,123 @@ def execute(self) -> ExchangeGenerator[None]:
         assert ex.line == "OK"
 
 
-class GetBlockNumbers(Command[Dict[str, int]]):
-    """Get the name and number of each block type in a dictionary,
-    alphabetically ordered
+@dataclass
+class GetBlockInfo(Command[Dict[str, BlockInfo]]):
+    """Get the name, number, and description of each block type
+    in a dictionary, alphabetically ordered
+
+    Args:
+        skip_description: If `True`, prevents retrieving the description
+            for each Block. This will reduce network calls.
 
     For example::
 
-        GetBlockNumbers() -> {"LUT": 8, "PCAP": 1, ...}
+         GetBlockInfo() ->
+             {
+                 "LUT": BlockInfo(number=8, description="Lookup table"),
+                 "PCAP": BlockInfo(number=1, description="Position capture control"),
+                 ...
+             }
     """
 
-    def execute(self) -> ExchangeGenerator[Dict[str, int]]:
+    skip_description: bool = False
+
+    def execute(self) -> ExchangeGenerator[Dict[str, BlockInfo]]:
         ex = Exchange("*BLOCKS?")
         yield ex
-        blocks_list = []
+
+        blocks_list, commands = [], []
         for line in ex.multiline:
             block, num = line.split()
             blocks_list.append((block, int(num)))
-        return dict(sorted(blocks_list))
+            commands.append(Get(f"*DESC.{block}"))
+
+        if self.skip_description:
+            description_values = tuple(None for _ in range(len(commands)))
+        else:
+            description_values = yield from _execute_commands(*commands)
+
+        blocks_info = {
+            block[0]: BlockInfo(number=block[1], description=desc)
+            for block, desc in zip(blocks_list, description_values)
+        }
+
+        return OrderedDict(sorted(blocks_info.items()))
 
 
 @dataclass
-class GetFields(Command[Dict[str, FieldType]]):
-    """Get the fields of a block, returning a `FieldType` for each one, ordered
+class GetFieldInfo(Command[Dict[str, FieldInfo]]):
+    """Get the fields of a block, returning a `FieldInfo` for each one, ordered
     to match the definition order in the PandA
```

Comment:
Question: The documentation here has always said that GetFields/GetFieldInfo will return the fields ordered as they are in the PandA. This appears to be wrong - the code does a `sorted()` call on the list so that they are returned in alphabetical order. And we can see that the PandA often returns its fields in non-alphabetical order. So which should it be?